bin/create_hugo_pages.py

```diff
@@ -67,7 +70,114 @@ def check_directory(cdir, clean=False):
+def month_to_number(month):
+    """Convert a month name to a number.
+
+    Handles misspelled and abbreviated months by checking for longest
+    match from start (>=2).
+
+    Args:
+        month: month name
+
+    Returns:
+        month number (1-12)
+    """
+    monthnames = [
+        "january",
+        "february",
+        "march",
+        "april",
+        "may",
+        "june",
+        "july",
+        "august",
+        "september",
+        "october",
+        "november",
+        "december",
+    ]
+    common = []
+    # handle misspelled and abbreviated months by checking for longest match from start (>=2)
+    for mi, m in enumerate(monthnames):
+        i = 0
+        while m[i] == month[i].lower() and i < len(month) - 1:
+            i += 1
+        common.append((mi, i))
+    return max(common, key=lambda x: x[1])[0] + 1
+
+
+def construct_date(year, month):
+    """Construct a Hugo date from a year and month.
+
+    Can handle the following month formats:
+
+    - D-D month (e.g. 1-4 January)
+    - month D-D (e.g. January 1-4)
+    - D month (e.g. 1 January)
+    - month D (e.g. January 1)
+    - month (e.g. January)
+    - month D - month D (e.g. January 30 - February 2)
+    - D month - D month (e.g. January 30 - February 2)
+    - M (e.g. 7)
+    - variants of the above with em and en dashes or multiple dashes
+    - variants of the above with capitalized or lower case month names
+
+    Args:
+        year: year field as present in the XML base data
+        month: month field as present in the XML base data
+    """
+    if not re.fullmatch(r"\d{4}", year):
+        raise Exception("Invalid year: {}".format(year))
+    y = int(year)
+    dashes = r"(?:[—–-]+|/|and|to)"
+    monthnames = [
+        "january",
+        "february",
+        "march",
+        "april",
+        "may",
+        "june",
+        "july",
+        "august",
+        "september",
+        "october",
+        "november",
+        "december",
+    ]
+    fullmonth = f"({'|'.join(monthnames)})"
+    fullmonth = r"\b([a-zA-Z]{2,})\b"
+    day = r"(\d{1,2})(?:st|nd|rd|th)?"
+    numericmonth = r"(\d{1,2})"
+    if re.fullmatch(numericmonth, month):
+        m = int(month)
+        return f"{y:04d}-{m:02d}-01"
+    if re.fullmatch(fullmonth, month, re.IGNORECASE):
+        m = month_to_number(month)
+        return f"{y:04d}-{m:02d}-01"
+    d = None
+    m = None
+    match = re.search(r"\s*".join([day, dashes, day]), month)
+    if match:
+        d, _ = match.groups()
+    else:
+        parts = re.split(dashes, month)
+        if len(parts) == 2:
+            month = parts[0]
+    match = re.search(day, month)
+    if d is None and match:
+        d = match.group(1)
+    match = re.search(fullmonth, month, re.IGNORECASE)
+    if match:
+        m = month_to_number(match.group(1))
+    if m is None:
+        log.error(f"Could not parse month: {month}")
+        m = 1
+    if d is None:
+        d = 1
+    return f"{y:04d}-{m:02d}-{int(d):02d}"
```

Comment:
+1 that we should remove this code if we're not planning on using it.

Reply:
Done. :+1: